cdap-elastic/src/main/java/co/cask/cdap/metadata/elastic/ElasticsearchMetadataStorage.java

```diff
@@ -661,7 +667,7 @@ private void setRefreshPolicy(WriteRequest<?> request) {
     SearchHits hits = searchResponse.getHits();
     List<MetadataRecord> results = fromHits(hits);
     String newCursor = computeCursor(searchResponse, request.getOffset(), request.getLimit());
```

Comment:
Any reason why `request.getLimit()` is used here vs the parameter `limit` in other places of this method.

Reply:
good question. I think you are right and it should use limit and offset instead. Let me add a test for that

Reply:
actually they are identical, because search() passes on request.getOffset/Limit() as offset and limit. added validation to the test, anyway.